lib/puppet_metadata/beaker.rb

```diff
@@ -52,20 +52,31 @@ def adjusted_os(os)
       #   Enforce a domain to be appended to the hostname, making it an FQDN
       # @param [Optional[String]] puppet_version
       #   The desired puppet version. Will be appended to the hostname
+      # @param [Optional[String]] hostname
+      #   Override the automatically generated hostname.  The domain may still
+      #   be set via the `domain` param.
       #
       # @return [nil] If no setfile is available
       # @return [Array<(String, String)>] The beaker setfile description with a readable name
-      def os_release_to_setfile(os, release, use_fqdn: false, pidfile_workaround: false, domain: nil, puppet_version: nil)
+      def os_release_to_setfile(os, release, use_fqdn: false, pidfile_workaround: false, domain: nil, puppet_version: nil, hostname: nil)
         return unless os_supported?(os)
 
         aos = adjusted_os(os)
-
         name = "#{aos}#{release.tr('.', '')}-64"
-        hostname = (puppet_version.nil? && puppet_version != 'none') ? name : "#{name}-#{puppet_version}"
+        shostname = if hostname.nil?
+                      (puppet_version.nil? && puppet_version != 'none') ? name : "#{name}-#{puppet_version}"
+                    else
+                      hostname
+                    end
```

Comment:
I think this would be easier to follow
```suggestion
        shostname = if hostname
                      hostname
                    elsif !puppet_version.nil? && puppet_version != 'none'
                      "#{name}-#{puppet_version}"
                    else
                      name
                    end
```

After reading that, I think it raises the question: should it always append the Puppet version if available? So essentially use:
```ruby
shostname = hostname || name
shostname += "-#{puppet_version}" if !puppet_version.nil? && puppet_version != 'none'
```

Reply:
I have no objection to that.  This is puppet specific machinery.

Reply:
I test commited the suggestion and it breaks existing tests as `-none` no longer appears as the puppet version.  It seems like the existing code is already forcing `"#{name}-#{puppet_version}"` because of a logic bug.

Reply:
This logic is wrong as it will append the puppet version to the output hostname even when the `hostname` param is set, which isn't the intention.  We need to be able to set the hostname to use in software configuration to communicate between hosts.

```
shostname = hostname || name
shostname += "-#{puppet_version}" if !puppet_version.nil? && puppet_version != 'none'
```

Reply:
This seems to fix the existing tests: 
```
         shostname = if hostname
                       hostname
-                    elsif !puppet_version.nil? && puppet_version != 'none'
+                    elsif !puppet_version.nil?
                       "#{name}-#{puppet_version}"
                     else
                       name
```